### PR TITLE
Add Form Validation to Contact Form

### LINKED ISSUE #4: Add contact form validation
The contact form should require that there is some text in both the email and name fields, the message should be optional.
Email validation is notoriously unreliable so let's just require that something was entered.  That way if someone decides they want to put their mastodon handle in there that would be ok as well.

### PR DESCRIPTION
**Pull request by shaanardev**
Resolve #4 by:
-  Adds validation to the form by checking that there is a value for the Name and Email input on submit.
-  Adds the following error handling and supporting UX:
     -  Generates error message on form validation for which **required** form inputs were invalid or empty.
     -  Displays error message with red background when form is invalid.
     -  Error message disappears when user is typing controlled by `createEventDispatcher` in `TextInput` component.

### CODE REVIEW

File: src/components/contactForm/ContactForm.svelte [hunk 1]
```diff
@@ -1,39 +1,133 @@
...
+			/>
+		{/if}
+
+		{#if formItem.formType === FormType.TEXTBOX}
+			<TextBoxInput fieldName="Message" bind:value={formItem.itemValue} />
+		{/if}
+	{/each}
+
+	{#if !isUserTyping && !isFormValid}
+		<p class={!isFormValid ? 'error' : 'error__hide'}>{errorMessage}</p>
```
Comment (KleeUT) [inline src/components/contactForm/ContactForm.svelte:128]: I'd prefer if we were to show the error per field, close to the field.  Maybe something more like this?
Error messages

File: src/components/contactForm/ContactForm.svelte [hunk 2]
```diff
@@ -69,4 +163,17 @@
 		color: black;
 		cursor: not-allowed;
 	}
+
+	.error {
+		display: block;
+		background-color: crimson;
```
Comment (KleeUT) [inline src/components/contactForm/ContactForm.svelte:169]: This has a bit of a low contrast ratio.  https://contrast-ratio.com/#black-on-crimson
Using a light background and dark text colour shows up better https://contrast-ratio.com/#%236b0000-on-%23ffefef

**Review by KleeUT — CHANGES_REQUESTED** [review]
Thanks for your contribution.  I've left a few comments in the code.

**Comment by shaanardev** [comment]
Feedback addressed in latest commit, thank you.

**Review by KleeUT — APPROVED** [review]

**Pull request merged**